state/account.go

```diff
@@ -93,13 +129,77 @@ func (st *Account) Deserialize(buf []byte) error {
 	return nil
 }
 
+// IsNewbieAccount returns true if the account has not sent any actions
+func (st *Account) IsNewbieAccount() bool {
+	return st.nonce == 0
+}
+
+// AccountType returns the account type
+func (st *Account) AccountType() int32 {
+	return st.accountType
+}
+
+// SetPendingNonce sets the pending nonce
+func (st *Account) SetPendingNonce(nonce uint64) error {
```

Comment:
IMO, we should keep using `SetNonce(nonce)`, this way `SetPendingNonce(nonce+1)` feels  more confusing